### PR TITLE
Add an early version of the documentation

### PR DESCRIPTION
**Pull request by qbarrand**
This will be easier to track than the HackMD thing. Comments welcome.

### CODE REVIEW

File: docs/platforms.md [hunk 1]
```diff
@@ -0,0 +1,6 @@
+# Platforms
+
+`r2pm` supports installing packages on the following operating systems, all running the `amd64` architecture:
+- Windows 7 and later
+- Linux
+- macOS 10.14+
```
Comment (XVilka) [inline docs/platforms.md]: FreeBSD also?

File: docs/packages.md [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+
+  linux:
+    source:
+      type: git
+      repo: git@github.com:username/project.git
+      ref: master  # or a tag
+    commands:
+      - './configure --prefix {{ .DestPath }}'
+      - make
+      - make install  # avoid this, as r2pm will not manage those files unless they are also specified in the 'out' section
```
Comment (Maijin) [inline docs/packages.md]: If it has to be avoided, don't put it there.

File: docs/packages.md [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+version: 1.2.3
+description: Some description   
+
+install:
+
+  windows:
+    source:
+      type: zip
+      url: http://a-random.url/zip-archive.zip
+    # commands: []  # Unlikely for Windows
```
Comment (Maijin) [inline docs/packages.md]: remove the "# Unlikely for Windows"

**Comment by XVilka** [comment]
Please update and we'll merge. Thank you!

**Pull request merged**